src/snowflake/snowpark/_internal/data_source/datasource_partitioner.py

```diff
@@ -95,6 +95,12 @@ def schema(self) -> StructType:
                 self.table_or_query, self.is_query, self._query_input_alias
             )
         else:
+            self.driver.get_raw_schema(
+                self.table_or_query,
+                self.driver.create_connection().cursor(),
+                self.is_query,
+                self._query_input_alias,
+            )
```

Comment:
could you remind me what's the issue and why getting raw schema from DBMS could fix the issue?

Reply:
we hitting the issue because some DBMS need raw type of DBMS to compose the right select query.
getting raw schema would satisfy that need